### PR TITLE
enable task in Python

### PR DESCRIPTION
**Pull request by yu239**
Now the user can define a task given a map in Python. However, the environment is still configured in C++.

### CODE REVIEW

File: games/xworld/tasks/XWorldNavTargetPy.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import random
+
+"""
+Each stage function has two inputs:
+
+entity:
```
Comment (Haichao-Zhang) [inline games/xworld/tasks/XWorldNavTargetPy.py]: list of entities?

File: games/xworld/tasks/XWorldNavTargetPy.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  id        - unique str for this entity
+  type      - "agent", "goal", "block"
+  location  - (x, y, z)
+  (optional:)
+  name      - name of the entity
+  color     - color of the entity
+
+dimensions:
```
Comment (Haichao-Zhang) [inline games/xworld/tasks/XWorldNavTargetPy.py]: dimensions of the world?

File: games/xworld/tasks/XWorldNavTargetPy.py (hunk 1 above)
Comment (yu239) [reply games/xworld/tasks/XWorldNavTargetPy.py]: Comments added

File: games/xworld/tasks/XWorldNavTargetPy.py (hunk 2 above)
Comment (yu239) [reply games/xworld/tasks/XWorldNavTargetPy.py]: Comments added

**Pull request merged**